### PR TITLE
Query processor IPA protocol end-to-end implementation

### PR DESCRIPTION
**Pull request by akoshelev**
A bunch of boilerplate code had to be written to serialize and deserialize structs. that's quite annoying, need to think about a better way how to deal with it.

### CODE REVIEW

File: src/protocol/attribution/mod.rs [hunk 1]
```diff
@@ -40,6 +41,52 @@ pub struct AggregateCreditOutputRow<F: Field> {
     credit: Replicated<F>,
 }
 
+impl<F: Field> AggregateCreditOutputRow<F> {
+    // TODO: this gets repetitive. For the most of our structures we need to be able to serialize
```
Comment (thurstonsand) [inline src/protocol/attribution/mod.rs:45]: yeah this kinda feels like you want to recreate the `serde::Serialize/Deserialize` macros...

File: src/protocol/ipa/mod.rs [hunk 2]
```diff
@@ -75,13 +76,65 @@ impl AsRef<str> for IPAInputRowResharableStep {
...
 
+impl<F: Field> IPAInputRow<F> {
+    pub const SIZE_IN_BYTES: usize = std::mem::size_of::<Self>();
+
+    /// Splits the given slice into chunks aligned with the size of this struct and returns an
+    /// iterator that produces deserialized instances.
+    ///
+    /// ## Panics
+    /// Panics if the slice buffer is not aligned with the size of this struct.
+    pub fn from_byte_slice(input: &[u8]) -> impl Iterator<Item = Self> + '_ {
```
Comment (thurstonsand) [inline src/protocol/ipa/mod.rs:95]: no `deserialize` method here?

File: src/query/executor.rs [hunk 3]
```diff
@@ -66,11 +91,25 @@ async fn test_multiply<F: Field, S: Stream<Item = Vec<u8>> + Send + Unpin>(
 }
 
 #[allow(clippy::unused_async)]
```
Comment (thurstonsand) [inline src/query/executor.rs:93]: remove

**Review by thurstonsand — APPROVED** [review]
yep, mostly just serde and hooking up the tests

File: src/protocol/ipa/mod.rs (hunk 2 above)
Comment (akoshelev) [reply src/protocol/ipa/mod.rs:95]: for now no, because we only deserialize many of them

**Pull request merged**